classes/order/Order.php

```diff
@@ -1848,11 +1848,35 @@ public function addOrderPayment($amount_paid, $payment_method = null, $payment_t
             $order_payment->date_add .= ' ' . date('H:i:s');
         }
 
+        /*
+         * 4 cases
+         *
+         * Order is in default_currency + Payment is in Order currency
+         *    for example default = 1, order = 1, payment = 1
+         *    ==> NO conversion to do
+         * Order is in default_currency + Payment is NOT in Order currency
+         *    for example default = 1, order = 1, payment = 2
+         *    ==> convert payment in order's currency
+         * Order is NOT in default_currency + Payment is in Order currency
+         *    for example default = 1, order = 2, payment = 2
+         *    ==> NO conversion to do
+         * Order is NOT in default_currency + Payment is NOT in Order currency
+         *    for example default = 1, order = 2, payment = 3
+         *    ==> As conversion rates are set regarding the default currency,
+         *        convert payment to default and from default to order's currency
+         */
+
         // Update total_paid_real value for backward compatibility reasons
         if ($order_payment->id_currency == $this->id_currency) {
             $this->total_paid_real += $order_payment->amount;
         } else {
-            $this->total_paid_real += Tools::ps_round(Tools::convertPrice($order_payment->amount, $order_payment->id_currency, false), 2);
+            $default_currency = (int) Configuration::get('PS_CURRENCY_DEFAULT');
+            if ($this->id_currency === $default_currency) {
+                $this->total_paid_real += Tools::ps_round(Tools::convertPrice($order_payment->amount, $this->id_currency, false), 2);
```

Comment:
The precision should not be to the default currency precision ?

Reply:
Indeed the precision should be related to the currency However I don't think we need to deal with it in this PR's scope
Because there are actually MANY places where precision is not managed correctly, a PR should handle all these rounding at once

But it's worth question in however, in case of conversion which currency's precision should we use? I'd say it should be the "destination" precision And the rounding should only be performed in the last operation
Because if we perform several conversion we might lose some precision during the different computing (especially if one of the currency has a 0 precision..)